src/controllers/hid/hidioreport.cpp

```diff
@@ -0,0 +1,117 @@
+    }
+    m_latchedOutputReportData.replace(0, data.size(), data);
+    m_possiblyUnsendDataLatched = true;
+}
+
+bool HidIoReport::sendOutputReport(QMutex* pHidDeviceMutex,
+        hid_device* pHidDevice,
+        const mixxx::hid::DeviceInfo& deviceInfo,
+        const RuntimeLoggingCategory& logOutput) {
+    auto startOfHidWrite = mixxx::Time::elapsed();
+    QByteArray reportToSend;
+
+    auto lock = lockMutex(pHidDeviceMutex);
+
+    {
+        auto lock = lockMutex(&m_latchedOutputReportDataMutex);
+
+        if (!m_possiblyUnsendDataLatched) {
+            // Return with false, to signal the caller, that no time consuming IO operation was necessary
+            return false;
+        }
+
+        if (!m_lastSentOutputReportData.compare(m_latchedOutputReportData)) {
+            // An HID OutputReport can contain only HID OutputItems.
+            // HID OutputItems are defined to represent the state of one or more similar controls or LEDs.
+            // Only HID Feature items may be attributes of other items.
+            // This means there is always a one to one relationship to the state of control(s)/LED(s),
+            // and if the state is not changed, there's no need to execute the time consuming hid_write again.
+            qCDebug(logOutput) << "t:" << startOfHidWrite.formatMillisWithUnit()
+                               << " Skipped identical Output Report for"
+                               << deviceInfo.formatName() << "serial #"
+                               << deviceInfo.serialNumberRaw() << "(Report ID"
+                               << m_reportId << ")";
+
+            // Setting m_possiblyUnsendDataLatched to false prevents, that the byte array compare operation is executed for the same data again
+            m_possiblyUnsendDataLatched = false;
+
+            // Return with false, to signal the caller, that no time consuming IO operation was necessary
+            return false;
+        }
+
+        // hid_write requires the first byte to be the Report ID, followed by the data[] to be send
+        reportToSend.reserve(m_latchedOutputReportData.size() + kReportIdSize);
+        reportToSend.append(m_reportId);
+        reportToSend.append(m_latchedOutputReportData);
+
+        // Preemtive set m_lastSentOutputReportData and m_possiblyUnsendDataLatched,
+        // to release the mutex during the time consuming hid_write operation.
+        // In the unlikely case that hid_write fails, they will be invalidated afterwards
+        // This is safe, because these members are only reset in this scope of this method,
+        // and concurrent execution of this method is prevented by locking pHidDeviceMutex
+        m_lastSentOutputReportData.swap(m_latchedOutputReportData);
+        m_possiblyUnsendDataLatched = false;
+    }
+
+    // hid_write can take several milliseconds, because hidapi synchronizes the asyncron HID communication from the OS
+    int result = hid_write(pHidDevice,
```

Comment:
The scope of pHidDeviceMutex should be restricted to this call only.

Reply:
That would be a bit too restricted - hid_write and the succeeding hid_error belong together. No other hidapi command (for this device) must be executed inbetween.

Reply:
I didn't notice the second call. Is it possible to keep the locking regions disjunct? This is the crucial point.

Reply:
No, the hidapi structure where pHidDevice points to, stores the error message of hid_write, which hid_error prints. If an concurrent thread would execute another hid operation, the error message of the wrong operation could be printed here.

Reply:
There is no overlapping locking of different mutexes anymore